include/options.hxx

```diff
@@ -105,76 +150,255 @@ using std::string;
 class Options {
 public:
   /// Constructor. This is called to create the root object
-  Options();
+  Options() : parent_instance(nullptr) {}
```

Comment:
`parent_instance` has a default value, so we can skip setting it in the ctor. But it doesn't do any harm